asapdiscovery-docking/asapdiscovery/docking/docking_v2.py

```diff
@@ -0,0 +1,283 @@
+                pose_res, retcode = self.run_oe_posit_docking(
+                    opts, pose_res, du, lig_oemol, self.num_poses
+                )
+
+            # try again allowing clashes
+            if (
+                self.allow_final_clash
+                and ret_code == oedocking.OEDockingReturnCode_NoValidNonClashPoses
+            ):
+                opts.SetPoseRelaxMode(oedocking.OEPoseRelaxMode_ALL)
+                opts.SetAllowedClashType(oedocking.OEAllowedClashType_ANY)
+                pose_res, retcode = self.run_oe_posit_docking(
+                    opts, pose_res, du, lig_oemol, self.num_poses
+                )
+
+            if ret_code == oedocking.OEDockingReturnCode_Success:
+                for result in pose_res.GetSinglePoseResults():
+                    posed_mol = result.GetPose()
+                    prob = result.GetProbability()
+                    pose_scorer = oedocking.OEScore(oedocking.OEScoreType_Chemgauss4)
+                    pose_scorer.Initialize(du)
+                    chemgauss_score = pose_scorer.ScoreLigand(posed_mol)
+
+                    posed_ligand = Ligand.from_oemol(posed_mol, **pair.ligand.dict())
+                    # set SD tags
+                    sd_data = {
+                        DockingResultCols.DOCKING_CONFIDENCE_POSIT.value: prob,
+                        DockingResultCols.DOCKING_SCORE_POSIT.value: chemgauss_score,
+                        DockingResultCols.POSIT_METHOD.value: POSIT_METHOD.reverse_lookup(
+                            self.posit_method.value
+                        ),
+                    }
+                    posed_ligand.set_SD_data(sd_data)
+
+                    docking_result = DockingResult(
+                        input_pair=pair,
+                        posed_ligand=posed_ligand,
+                        probability=prob,
+                        score=chemgauss_score,
+                        score_type=self.score_type,
+                        provenance=self.provenance(),
+                    )
+                    docking_results.append(docking_result)
+
+                    if self.write_files:
+                        # write out the docked pose
+                        if names_unique:
+                            output_dir = self.output_dir / lig.compound_name
+                        else:
+                            output_dir = self.output_dir / f"unknown_ligand_{i}"
+                        output_dir.mkdir(parents=True, exist_ok=True)
+                        output_sdf_file = output_dir / "docked.sdf"
+                        output_pdb_file = output_dir / "docked_complex.pdb"
+
+                        posed_ligand.to_sdf(output_sdf_file)
+
+                        combined_oemol = docking_result.to_posed_oemol()
+                        save_openeye_pdb(combined_oemol, output_pdb_file)
+
+            else:
+                pass
+
+        return docking_results
+
+    def provenance(self) -> dict[str, str]:
```

Comment:
do we want to also include the docking parameters/method used? or is that something for the future?

Reply:
Future, I will work on all the provenance stuff down the track.